### PR TITLE
[mypyc] Faster set creation with generator expression

### PR DESCRIPTION
**Pull request by 97littleleaf11**
### Description

Closes https://github.com/mypyc/mypyc/issues/771 and https://github.com/python/mypy/pull/9707

Changes:
* Move code from `tranform_set_comprehension` to `translate_set_comprehension`
* Fix the unnecessary set creation (https://github.com/python/mypy/pull/9707#issuecomment-725549735)

## Test Plan

Both generated IR test and functional test:
* Set built from iterators (`list` `tuple` `dict` `range`)
* Set built from nested generator

### CODE REVIEW

**Comment by JukkaL** [comment]
Thanks for the PR! Have you benchmarked this against master? (If not, no worries, I can do some quick benchmarking tomorrow.)

**Comment by 97littleleaf11** [comment]
> Thanks for the PR! Have you benchmarked this against master? (If not, no worries, I can do some quick benchmarking tomorrow.)

Not yet.

File: mypyc/irbuild/specialize.py [hunk 1]
```diff
@@ -143,6 +143,9 @@ def translate_safe_generator_call(
                 val = tuple_from_generator_helper(builder, expr.args[0])
                 if val is not None:
                     return val
+            if callee.fullname == "builtins.set":
+                return translate_set_comprehension(builder, expr.args[0])
+
```
Comment (97littleleaf11) [inline mypyc/irbuild/specialize.py:148]: Maybe I should move `builtins.set` to a new specialize function ?

Comment (JukkaL) [reply mypyc/irbuild/specialize.py:148]: That sounds reasonable, since the implementation isn't sharing much with the other cases.

**Comment by JukkaL** [comment]
I measured the performance and now the speed of calling `set` with a generator expression is the same as a set comprehension. Nice! Calling `set` was 62% faster in the ad-hoc benchmark that I used, compared to master.

**Review by JukkaL — APPROVED** [review]
Thanks, looks good! This is a nice performance win.

**Pull request merged**